### PR TITLE
refactor: steps support css var

### PR DESCRIPTION
**Pull request by cc-hearts**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- [https://github.com/ant-design/ant-design/issues/45618](https://github.com/ant-design/ant-design/issues/45618)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Steps support css variables     |
| 🇨🇳 Chinese |     步骤条支持css变量      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at c286535</samp>

This pull request refactors the steps component to use the `@ant-design/cssinjs` package and a custom hook for CSS variable support. It improves the compatibility, consistency, and customizability of the component style and fixes some layout and unit issues. It also removes some redundant code and adds new properties and functions to the component token interface.

### <samp>🤖 Generated by Copilot at c286535</samp>

*  Add the `useCSSVar` hook and the `wrapCSSVar` function to provide CSS variables for the steps component ([link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9c24aa194a67d1070caa8793ae4ddda9d8fcb235f68cda41d58098eb3103172bR1-R4), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-a9a9cfe5e51c2d313f50fd8b15ef2d2723b06b9992d42063cf1ebe97023d1968R17), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-a9a9cfe5e51c2d313f50fd8b15ef2d2723b06b9992d42063cf1ebe97023d1968L82-R84), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-a9a9cfe5e51c2d313f50fd8b15ef2d2723b06b9992d42063cf1ebe97023d1968L130-R132))
*  Import and use the `unit` function from the `@ant-design/cssinjs` package to convert numbers to CSS unit strings in various style modules ([link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-37cad01c16ee3dabe57d26019d6a684f6b1c306e9f624c56a893c2b0159661f6R3), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-7c302bb80b5f306d72c03ca5e5fe2961248a1cdf0f41329ce9e3307110aa29d6L1-R4), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL1-R1), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-01b2e4af91ccc735ec0d4cb440d319986e58a6927f9ef0cd4a5bbd200e71bd08L1-R1), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-e7d8c4a1a9b938afa3709335c6eece6df2ea56049221a1ae1bd3221bd050a04eL1-R1), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9691ddb57044c160227ea61f46d3c85bbdf96606a3343a7faccd75d29b164ec4L1-R1),  [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-631448e52d62c21292b8a84d981ff17b265be188ad7d3428f5e3afad48d2a061L1-R1), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-c62b6f365b7960e69869146b451e82a05ffff0850ce4b3193bb8b33224ca9f01L1-R1))
*  Add the `tailTop` property to the `ComponentToken` interface and calculate its value based on the `dotSize` and `lineWidth` ([link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-7c302bb80b5f306d72c03ca5e5fe2961248a1cdf0f41329ce9e3307110aa29d6R82-R86))
*  Use the `calc` and `equal` methods of the token to calculate various style properties based on the token values, such as `width`, `height`, `top`, `margin`, `padding`, `border`, etc. ([link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-7c302bb80b5f306d72c03ca5e5fe2961248a1cdf0f41329ce9e3307110aa29d6L227-R232), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-7c302bb80b5f306d72c03ca5e5fe2961248a1cdf0f41329ce9e3307110aa29d6L247-R256), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL23-R24), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL39-R44), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL50-R50), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL57-R57), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL65-R65), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL88-R88), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL98-R98), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-ad748dfbee3431d68a354ef484095347173ef0367c8e4c5f830469150d7e056bL107-R107), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-01b2e4af91ccc735ec0d4cb440d319986e58a6927f9ef0cd4a5bbd200e71bd08L14-R20), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-01b2e4af91ccc735ec0d4cb440d319986e58a6927f9ef0cd4a5bbd200e71bd08L49-R54), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-e7d8c4a1a9b938afa3709335c6eece6df2ea56049221a1ae1bd3221bd050a04eL22-R22), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-e7d8c4a1a9b938afa3709335c6eece6df2ea56049221a1ae1bd3221bd050a04eL34-R34), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-e7d8c4a1a9b938afa3709335c6eece6df2ea56049221a1ae1bd3221bd050a04eL74-R82), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-e7d8c4a1a9b938afa3709335c6eece6df2ea56049221a1ae1bd3221bd050a04eL119-R120), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-e7d8c4a1a9b938afa3709335c6eece6df2ea56049221a1ae1bd3221bd050a04eL127-R128), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9691ddb57044c160227ea61f46d3c85bbdf96606a3343a7faccd75d29b164ec4L23-R32), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9691ddb57044c160227ea61f46d3c85bbdf96606a3343a7faccd75d29b164ec4L39-R41), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9691ddb57044c160227ea61f46d3c85bbdf96606a3343a7faccd75d29b164ec4L56-R62), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9691ddb57044c160227ea61f46d3c85bbdf96606a3343a7faccd75d29b164ec4L71-R84), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9691ddb57044c160227ea61f46d3c85bbdf96606a3343a7faccd75d29b164ec4L87-R102), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9691ddb57044c160227ea61f46d3c85bbdf96606a3343a7faccd75d29b164ec4L100-R116), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-9691ddb57044c160227ea61f46d3c85bbdf96606a3343a7faccd75d29b164ec4L112-R129), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-5dc058643c8ba29a6ae55b5c4c7692498b07f8be44092efcbecd160d3dd017a7L23-R28), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-5dc058643c8ba29a6ae55b5c4c7692498b07f8be44092efcbecd160d3dd017a7L36-R51), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-5dc058643c8ba29a6ae55b5c4c7692498b07f8be44092efcbecd160d3dd017a7L50-R79), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-631448e52d62c21292b8a84d981ff17b265be188ad7d3428f5e3afad48d2a061L30-R32), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-631448e52d62c21292b8a84d981ff17b265be188ad7d3428f5e3afad48d2a061L39-R42), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-631448e52d62c21292b8a84d981ff17b265be188ad7d3428f5e3afad48d2a061L50-R50), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-631448e52d62c21292b8a84d981ff17b265be188ad7d3428f5e3afad48d2a061L61-R61), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-c62b6f365b7960e69869146b451e82a05ffff0850ce4b3193bb8b33224ca9f01L25-R29), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-c62b6f365b7960e69869146b451e82a05ffff0850ce4b3193bb8b33224ca9f01L38-R43), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-c62b6f365b7960e69869146b451e82a05ffff0850ce4b3193bb8b33224ca9f01L62-R70))
*  Add the `prepareComponentToken` function to return the default token values for the steps component based on the global token values, and use it in the `genComponentStyleHook` function ([link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-7c302bb80b5f306d72c03ca5e5fe2961248a1cdf0f41329ce9e3307110aa29d6R391-R407), [link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-7c302bb80b5f306d72c03ca5e5fe2961248a1cdf0f41329ce9e3307110aa29d6L447-R469))
*  Use the `customIconFontSize` to set the `lineHeight` property for the custom icon style ([link](https://github.com/ant-design/ant-design/pull/45878/files?diff=unified&w=0#diff-37cad01c16ee3dabe57d26019d6a684f6b1c306e9f624c56a893c2b0159661f6L19-R20))

### CODE REVIEW

**Comment by MadCcc** [comment]
Argos 里的 diff 看一下。这个改造要保证相关组件没有变化
https://app.argos-ci.com/ant-design/ant-design/builds/12964/64906562

**Comment by cc-hearts** [comment]
done

File: components/steps/style/index.tsx [hunk 1]
```diff
@@ -383,6 +388,27 @@ const genStepsStyle: GenerateStyle<StepsToken, CSSObject> = (token) => {
 };
 
 // ============================== Export ==============================
+export const prepareComponentToken: GetDefaultToken<'Steps'> = (token) => {
+  const dotSize = token.Steps?.dotSize ?? token.controlHeight / 4;
```
Comment (MadCcc) [inline components/steps/style/index.tsx:392]: 这里不要这么写，只要关注初始化就可以了，用户自定义的值在其他地方会注入

File: components/steps/style/index.tsx [hunk 2]
```diff
@@ -383,6 +388,27 @@ const genStepsStyle: GenerateStyle<StepsToken, CSSObject> = (token) => {
...
+
+  return {
+    titleLineHeight: token.controlHeight,
```
Comment (MadCcc) [inline components/steps/style/index.tsx:395]: 这个 token 需要

File: components/steps/style/index.tsx [hunk 3]
```diff
@@ -444,29 +470,5 @@ export default genComponentStyleHook(
 
     return [genStepsStyle(stepsToken)];
```
Comment (MadCcc) [inline components/steps/style/index.tsx:471]: 上面所有和 wireframe 相关的 token 都需要提升为 component token

Comment (MadCcc) [reply components/steps/style/index.tsx:471]: 具体可以对比一下 “线框风格” 这个 demo

File: components/steps/style/index.tsx (hunk 1 above)
Comment (cc-hearts) [reply components/steps/style/index.tsx:392]: 目前还没有想到如何处理 `Math.floor((dotSize - lineWidth * 3) / 2)` 中的 Math.floor. 能否给我点建议

File: components/steps/style/index.tsx [hunk 4]
```diff
@@ -383,23 +403,44 @@ const genStepsStyle: GenerateStyle<StepsToken, CSSObject> = (token) => {
 };
 
 // ============================== Export ==============================
+export const prepareComponentToken: GetDefaultToken<'Steps'> = (token) => {
+  const dotSize = token.Steps?.dotSize ?? token.controlHeight / 4;
```
Comment (MadCcc) [inline components/steps/style/index.tsx:407]:
```suggestion
  const dotSize = token.controlHeight / 4;
```

File: components/steps/style/index.tsx (hunk 1 above)
Comment (MadCcc) [reply components/steps/style/index.tsx:407]: 你的意思是 tailTop 不会跟着 dotSize 变化了？

**Comment by cc-hearts** [comment]
tailTop 需要随着 dotSize 变化。在 prepareComponentToken 不应该获取用户注入的值，但是在 prepareComponentToken 之外无法进行 Math.floor 的计算， 这是我目前遇到的困难

**Comment by cc-hearts** [comment]
tailTop 需要随着 dotSize 变化。
在 prepareComponentToken 不应该获取用户注入的值，但是在 prepareComponentToken 之外无法进行 Math.floor 的计算，这是我目前遇到的困难

**Comment by MadCcc** [comment]
这个 Math.floor 先去掉吧，就不提 component token 了。0.5px 是可以接受的，逻辑上也比较正确

**Comment by li-jia-nan** [comment]
argos diff 看起来是 Math.floor 导致的，应该是预期之内的吧

**Review by MadCcc — APPROVED** [review]

**Pull request merged**